packages/apps/esm-login-app/src/location-picker/location-picker.component.tsx

```diff
@@ -75,30 +76,33 @@ const LocationPicker: React.FC<LocationPickerProps> = ({ hideWelcomeMessage, cur
     state: LoginReferrer;
   };
 
+  const handleNavigation = useCallback(() => {
+    const referrer = state?.referrer;
+    const returnToUrl = new URLSearchParams(location?.search).get('returnToUrl');
+
+    if (referrer && !['/', '/login', '/login/location'].includes(referrer)) {
+      navigate({ to: '${openmrsSpaBase}' + referrer });
+      return;
+    }
+    if (returnToUrl && returnToUrl !== '/') {
```

Comment:
Even though this sort of is implied, I always think it's better to be explicit about this. This also allows us to skip the `return` on line 85.

```suggestion
    } else if (returnToUrl && returnToUrl !== '/') {
```